chapter/7/langs-consistency.md

```diff
@@ -1,11 +1,270 @@
+
+But where should these guarantees live? In the above Javascript example, the guarantees aren't guarantees at all, really. There's no restriction on what the programmer is allowed to do - the programmer has simply constructed a program that mirrors guarantees that she has modeled in her brain. In order to maintain properties such as *monotonicity*, she must constantly check the model in her brain against the code. We haven't really helped the programmer out that much - she has a lot of thinking to do.
+
+Databases such as PostgreSQL have issues like this as well, though they handle them quite differently, masters may need to ensure that write have occurred on every slave before the database becomes available for reading. A database system like this has pushed consistency concerns to the IO-level, completely out of the users control. They are enforced on system reads and system writes. This approach gives programmers no flexibility: as demonstrated with our shopping cart example, there's no need for these type of restrictions; we can tolerate inconsistency in order to maintain availability.
+
+Why not push the consistency guarantees in between the IO-level and the application-level? Is there any reason why you as the programmer couldn't program using tools that facilitate these types of monotonic programs? If you're familiar with formal systems -- why not construct a formal system (programming language / library) in which every theorem (program) is formally guarunteed to be monotonic? If it's *impossible* to express a non-monotonic program, the programmer needn't worry about maintaining a direct mapping between their code and their mental model.
+
+Wouldn't it be great if tools like this existed?
+
+### Bloom
+[ Introduce Bloom ]
+
+#### Restriction & Danger
+[Bloom restricts you, it's different, and it's dangerous]
+
+### Lasp
+[ Introduce Lasp ]
+Instead of trying to do it all (and accepting danger), it tries to be embeddable (and truly restrictive.)
+
+### Utilization
+
+Lasp is an Erlang library, and for good reason. Remember the initial discussion and reasoning for models such as Bloom and Lasp: we have a specific type of application that doesn't require tight consistency constraints. The constraints that do exist have been formalized, and we can be quite sure that by using a DSL like Lasp, we'll be safe from interns like Jerry. But Lasp can't do everything. More generally, eventual consistency doesn't solve every problem.
```

Comment:
Don't forget about the points that Chris & I made about Lasp being intended to be mixed and matched with regular Erlang code. That is, you don't need your entire program written in a language like Bloom (due to the analysis it provides). Instead you can write code such that it's mostly Erlang with a Lasp data structure or two.